scripts/Dockerfile.e2e

```diff
@@ -43,43 +43,46 @@ RUN apt-get update && \
         sed \
     && rm -rf /var/lib/apt/lists/*
 
+# Install clang
+ARG LLVM="llvmorg-16.0.4"
+ARG CLANG="clang+llvm-16.0.4-x86_64-linux-gnu-ubuntu-22.04"
+# NOTE: We `curl | tar` in spite of Docker best practices to save cache layers
+RUN curl -sL https://github.com/llvm/llvm-project/releases/download/${LLVM}/${CLANG}.tar.xz | \
+       tar -xvJ
+COPY src/glibc/wasi /${CLANG}/lib/clang/16/lib/wasi
+ENV PATH="/${CLANG}/bin:${PATH}"
+
+
 # Install rust and build wasmtime
 FROM base as build-wasmtime
 # NOTE: pinning known-to-work nightly-2025-06-01 (see #242)
 RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | \
     sh -s -- -y --default-toolchain nightly-2025-06-01
 ENV PATH="/root/.cargo/bin:${PATH}"
-COPY --parents src/wasmtime src/RawPOSIX src/fdtables src/sysdefs .
+COPY --parents src/wasmtime src/RawPOSIX src/fdtables src/sysdefs Makefile .
 # Build wasmtime with `--release` flag for faster tests
-# NOTE: `ln` workaround required for hard-coded paths in test tools (lind_config.sh)
-RUN cargo build --manifest-path src/wasmtime/Cargo.toml --release && \
-    (cd src/wasmtime/target && mkdir -p debug && ln -sf ../release/wasmtime debug)
+RUN make wasmtime
+
 
-# Install clang, build glibc and generate sysroot
+# Build glibc and generate sysroot
 FROM base AS build-glibc
-# NOTE: We `curl | tar` in spite of Docker best practices to save cache layers
-RUN curl -sL https://github.com/llvm/llvm-project/releases/download/llvmorg-16.0.4/clang+llvm-16.0.4-x86_64-linux-gnu-ubuntu-22.04.tar.xz | \
-       tar -xvJ
-COPY --parents scripts/make_glibc_and_sysroot.sh src/glibc .
-RUN ./scripts/make_glibc_and_sysroot.sh
+COPY --parents scripts/make_glibc_and_sysroot.sh src/glibc Makefile .
+RUN make sysroot
```

Comment:
Maybe using (and therefor COPYing) Makefile here (and in the build-wasm stage) is not such a good idea. Now, potentially unrelated changes in the Makefile will invalidate the cache layer and re-trigger the costly build. If this turns out to be a problem, we might want to revert to calling the build commands directly here. Should we add an issue, or code comment?